doc/whats-new.rst

```diff
@@ -66,6 +66,10 @@ New Features
 - Enable the limit option for dask array in the following methods :py:meth:`DataArray.ffill`, :py:meth:`DataArray.bfill`, :py:meth:`Dataset.ffill` and :py:meth:`Dataset.bfill` (:issue:`6112`)
   By `Joseph Nowak <https://github.com/josephnowak>`_.
 
+- Allow ``float`` ``n`` in :py:meth:`CFTimeIndex.shift` if ``freq`` is between ``D``
```

Comment:
I know the offsets are not currently public API, but your changes now are broader in scope than merely enabling `float` multiples in `shift`.  You've enabled multiplying tick offsets by floats, which I think is worth highlighting!